### ISSUE TITLE
A range of "*" cannot call intersects on ranges with "^" or "~"

### ISSUE THREAD

**Issue by jdalton** [npm/node-semver#236]
SemVer 5.5.0.
```js
new SemVer.Range("*").intersects(new SemVer.Range("3.0.19"))
// => true
new SemVer.Range("*").intersects(new SemVer.Range("^3.0.19"))
// TypeError: Invalid Version: [object Object]
```

**Update:**

If I change `"*"` to `">=0.0.0"` the `intersects` call will work.
```js
new SemVer.Range(">=0.0.0").intersects(new SemVer.Range("^3.0.19"))
// => true
```

**Comment by jdalton** [comment]
If I change `"*"` to `">=0.0.0"` the `intersects` call will work.
```js
new SemVer.Range(">=0.0.0").intersects(new SemVer.Range("^3.0.19"))
// => true
```

**Comment by pacamara** [comment]
@jdalton Hi! Have pushed a fix for this at https://github.com/npm/node-semver/pull/238 :beers:

**Comment by jdalton** [comment]
Thanks @pacamara!

**Comment by Asthetic** [comment]
I added my own solution as I was concerned that this wouldn't support 1.* or 1.0.*

https://github.com/npm/node-semver/pull/271

**Comment by isaacs** [comment]
Fixed in 6.1
